src/topology/algebra/uniform_ring.lean

```diff
@@ -150,8 +156,52 @@ instance : comm_ring (completion R) :=
       (assume a b, by rw [← coe_mul, ← coe_mul, mul_comm]),
  ..completion.ring }
 
-end uniform_space.completion
+@[simp] lemma map_mul_eq_mul_coe (r : R) :
+  completion.map ((*) r) = (*) (r : completion R) :=
+begin
+  ext x,
+  refine completion.induction_on x _ (λ s, _),
+  { exact is_closed_eq (completion.continuous_map) (continuous_mul_left (r : completion R)) },
+  { rw [map_coe (uniform_continuous_const_mul R r), coe_mul] },
+end
 
+instance : algebra R (completion R) :=
+{ commutes' := λ r x, by simp only [ring_hom.to_fun_eq_coe, mul_comm],
+  smul_def' := λ r x, congr_fun (map_mul_eq_mul_coe R r) x,
+  .. (uniform_space.completion.coe_ring_hom : R →+* completion R) }
+
+lemma algebra_map_eq_coe (r : R) : algebra_map R (completion R) r = coe r := rfl
+
+section algebra
+variables (S : Type*) [comm_semiring S] [algebra S R] [has_uniform_continuous_const_smul S R]
+
+@[simp] lemma map_smul_eq_mul_coe (s : S) :
+  completion.map ((•) s) = (*) (algebra_map S R s : completion R) :=
+begin
+  ext x,
+  refine completion.induction_on x _ (λ r, _),
+  { exact is_closed_eq (completion.continuous_map) (continuous_mul_left _) },
+  { rw [map_coe (has_uniform_continuous_const_smul.uniform_continuous_const_smul s),
+      algebra.smul_def, coe_mul];
+    apply_instance, },
+end
+
+instance algebra' : algebra S (completion R) :=
+{ commutes' := λ s x, by simp only [ring_hom.to_fun_eq_coe, mul_comm],
+  smul_def' := λ s x, congr_fun (map_smul_eq_mul_coe R S s) x,
+  ..((uniform_space.completion.coe_ring_hom : R →+* completion R).comp (algebra_map S R)) }
+
+@[simp] lemma algebra'_smul_eq (s : S) (x : completion R) :
+  s • x = (algebra_map S R s : completion R) * x :=
+by rw algebra.smul_def; refl
```

Comment:
I would recommend just deleting this lemma; if you really need it elsewhere, it's just `rw [algebra.smul_def, completion.algebra_map_def]`.